pandas/tests/groupby/test_transform.py

```diff
@@ -782,3 +782,14 @@ def test_any_all_np_func(func):
 
     res = df.groupby('key')['val'].transform(func)
     tm.assert_series_equal(res, exp)
+
+
+def test_transform_with_all_nan():
+    # GH 21624
+    df = DataFrame({'groups': [np.nan, np.nan, np.nan],
+                    'values': [1, 2, 3]})
+
+    grouped = df.groupby('groups')
+    result = grouped['values'].transform('sum')
+    expected = Series([np.nan, np.nan, np.nan], name='values')
```

Comment:
Hmm maybe I'm missing the point but shouldn't these be `6` and not `np.nan`?

Reply:
Currently, the behavior is to output `np.nan` for anything in a group with label `np.nan`. Making this output `6` would be changing the interpretation of `np.nan` from `not in any group` to `in a group with label np.nan`, which is going to be a pretty significant change in how things work. The treatment here of ignoring NA values is specified in the groupby docs. 

For example, see `test_groupby_transform_with_nan_group()` in `test_transform.py`